src/preflight/links-checks.js

```diff
@@ -62,56 +126,33 @@ export async function runLinksChecks(urls, scrapedObjects, context, options = {
         log.info('[preflight-audit] Found external links:', externalSet);
 
         // Check internal links
-        await Promise.all(
-          Array.from(internalSet).map(async (href) => {
-            try {
-              const response = await fetch(href, {
-                method: 'HEAD',
-                headers: {
-                  Authorization: options.pageAuthToken,
-                },
-                timeout: 3000,
-              });
-              const { status } = response;
-
-              if (status >= 400) {
-                log.debug(`[preflight-audit] internal url ${href} returned with status code: %s`, status);
-                brokenInternalLinks.push({ urlTo: href, href: pageUrl, status });
-              }
-            } catch (err) {
-              log.error(`[preflight-audit] Error checking internal link ${href} from ${pageUrl}:`, err.message);
-            }
+        const internalResults = await pMap(
+          Array.from(internalSet),
+          async (href) => checkLinkStatus(href, pageUrl, context, {
+            ...options,
+            isInternal: true,
           }),
+          { concurrency: 5 },
         );
```

Comment:
I would do sequential requests to avoid overloads using a reducer function for example:
```suggestion
const internalResults = await Array.from(internalSet).reduce(
  async (accPromise, href) => {
    const acc = await accPromise;

    const result = await checkLinkStatus(
      href,
      pageUrl,
      context,
      {
        ...options,
        isInternal: true,
      },
    );

    return [...acc, result];
  },
  Promise.resolve([])
);
```